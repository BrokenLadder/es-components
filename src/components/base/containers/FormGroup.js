import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './containers.less';

function FormGroup({
  hasError,
  inline,
  children
}) {
  const formClass = inline ? 'form-group-inline' : 'form-group';
  const classes = classNames(formClass, { error: hasError });

  return (
    <div className={classes}>
      {children}
    </div>
  );
}

FormGroup.propTypes = {
   /**
    * Should render in an error state
    */
  hasError: PropTypes.bool,
  /**
   * Should the child elements render line
   */
  inline: PropTypes.bool,
  children: PropTypes.element
};

FormGroup.defaultProps = {
  hasError: false,
  inline: false
};

export default FormGroup;
