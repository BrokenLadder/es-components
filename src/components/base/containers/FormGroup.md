Use a ``FormGroup`` to get proper spacing between form elements.

```
<div>
  <FormGroup>
    <label>
      <span className="label-style">First Name</span>
      <input type="text" />
    </label>
  </FormGroup>
  <FormGroup>
      <label>
        <span className="label-style">Last Name</span>
        <input type="text" />
      </label>
  </FormGroup>
</div>
```

A ``FormGroup`` can be rendered with errors as well.
```
<FormGroup hasError>
  <label>
    <span className="label-style">First Name</span>
    <input type="text" />
  </label>
</FormGroup>
```

Render a ``FormGroup`` with the inline prop set to true will render all its children in line with a 15 pixel margin between items.
```
<div>
  <FormGroup inline>
    <label>
      <span className="label-style">First Name</span>
      <input type="text" />
    </label>
    <label>
      <span className="label-style">Last Name</span>
      <input type="text" />
    </label>
  </FormGroup>
</div>
```
