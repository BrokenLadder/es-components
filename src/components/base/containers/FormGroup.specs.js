/* eslint-env jest */

import React from 'react';
import renderer from 'react-test-renderer';

import FormGroup from './FormGroup';

describe('FormGroup container component', () => {
  it('renders as expected', () => {
    const tree = renderer.create(
      <FormGroup>
        <div>Some child</div>
      </FormGroup>
    );

    expect(tree).toMatchSnapshot();
  });

  it('renders as expected when there is an error', () => {
    const tree = renderer.create(
      <FormGroup hasError>
        <div>Some child</div>
      </FormGroup>
    );

    expect(tree).toMatchSnapshot();
  });

  it('renders as expected when inline', () => {
    const tree = renderer.create(
      <FormGroup inline>
        <div>First child</div>
        <div>Second child</div>
      </FormGroup>
    );

    expect(tree).toMatchSnapshot();
  });
});
