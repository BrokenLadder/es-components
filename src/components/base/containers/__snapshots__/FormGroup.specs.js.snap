exports[`FormGroup container component renders as expected 1`] = `
<div
  className="biAlqo">
  <div>
    Some child
  </div>
</div>
`;

exports[`FormGroup container component renders as expected when inline 1`] = `
<div
  className="gOIrHN">
  <div>
    First child
  </div>
  <div>
    Second child
  </div>
</div>
`;

exports[`FormGroup container component renders as expected when there is an error 1`] = `
<div
  className="bSnWOL">
  <div>
    Some child
  </div>
</div>
`;
